src/dfx-core/src/config/model/canister_id_store.rs

```diff
@@ -189,6 +195,16 @@ impl CanisterIdStore {
         self.remote_ids
             .as_ref()
             .and_then(|remote_ids| self.get_name_in(canister_id, remote_ids))
+            .or_else(|| {
+                let principal = match Principal::from_str(canister_id) {
+                    Ok(p) => p,
+                    Err(_) => return None,
+                };
+                self.well_known_ids
+                    .iter()
+                    .find(|(_, id)| &&principal == id)
+                    .map(|(name, _)| name)
+            })
```

Comment:
If we do this here, before looking up the name in the project, then this is a potentially breaking change.  (maybe it is anyway).

Reply:
Agree fully, will move it to the end so it just adds on top and does not break anything previous.